framework/crud_resource_ops.go

```diff
@@ -7,6 +7,9 @@ import "context"
 // a guideline for an easier way to follow the rather complex intentions of
 // operators in general.
 type CRUDResourceOps interface {
+	// Name returns the resource's name used for identification.
+	Name() string
```

Comment:
After playing around I think it's better to keep `Name()` so it's more on par with `Resource` interface.